Mostly complete dashboard calendar

jrjohnson opened the pull request:
Has full filtering ability
View of school and user events
Agenda view / calendar view options
Separate events route

There are probably some rough edges, hopefully this meets enough of the requirements that we can merge and iterate for fixes.

Fixes #762, #761, #751, #750, #746,

Refs #206

dartajax commented:
Going back to 2014 with no calendar, the dates appear to all show the current date/time rather than the dates back in 2015 that should be displayed for the events.
![wrongdates](https://cloud.githubusercontent.com/assets/7493331/9420597/3213c694-481c-11e5-9aa4-16a4d0737164.JPG)

saschaben commented:
looks like attempting to toggle to the show "all events" throws a nice error:

```TypeError: Cannot read property 'get' of null```
``    at ilios.js:834``
``    at tryCatch (vendor.js:65653)``
``    at Object.invokeCallback (vendor.js:65668)``
``    at vendor.js:67453``
``    at vendor.js:42497``
``    at Queue.invoke (vendor.js:11476)``
``    at Object.Queue.flush (vendor.js:11540)``
``    at Object.DeferredActionQueues.flush (vendor.js:11340)``
``    at Object.Backburner.end (vendor.js:10733)``
``    at Object.Backburner.run (vendor.js:10801)```

Also, I think we haven't got the events list right; depending on the last location of the calendar view it show none, some, or some random list of events. This list should pay no attention to the calendar grid navigation, but should always default to load the list of current events, beginning with `NOW()` and scrolling into the future. It also should pay no nevermind to the filters. I think.

saschaben commented:
Also it looks from here like the topics, courses, cohorts, and session types filters never actually populate, throwing an error on page load

`TypeError: Cannot read property 'get' of null`
 `   at dashboard-calendar.js:247`
  `  at tryCatch (ember.debug.js:55078)`
   ` at Object.invokeCallback (ember.debug.js:55093)`
    `at ember.debug.js:56878`
    `at ember.debug.js:31922`
    `at Queue.invoke (ember.debug.js:901)`
    `at Object.Queue.flush (ember.debug.js:965)`
    `at Object.DeferredActionQueues.flush (ember.debug.js:765)`
    `at Object.Backburner.end (ember.debug.js:158)`
    `at Object.Backburner.run (ember.debug.js:226)`

saschaben commented:
...and my last comment before getting back to trying to have friday: in the header bar of the calendar filter box, the user name is not appearing for the french translation (it is doing so for spanish); for frenchies we just get "Calendrier de Monday, August 25 2015" instead of "Calendrier de Jon 'le dude' Johnson, August 25 2015". prolly a typo in the translation file.

jrjohnson commented:
New version.  Deals with the agenda view issues (date display, display this week no matter what).  Also added the academic year to control course and cohort lists.

I added the translation 'noEvents': 'Your schedule is empty this week.', for an empty agenda.  Is that the correct message?  and what is that in French and Spanish?

jrjohnson commented:
Updated to include new translations and to fix tests.  Please review.

thecoolestguy commented on app/components/toggle-yesno.js:
```diff
@@ -0,0 +1,9 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  lable: null,
```
Is 'lable' on purpose?  Should it not be 'label'?
